pkg/aws/actuator/actuator.go

```diff
@@ -326,47 +336,96 @@ func (a *AWSActuator) sync(ctx context.Context, cr *minterv1.CredentialsRequest)
 		logger.Debug("credentials already up to date")
 		return nil
 	}
-
-	credentialsRootSecret, err := a.GetCredentialsRootSecret(ctx, cr)
-	if err != nil {
-		logger.WithError(err).Error("issue with cloud credentials secret")
-		return err
+	stsDetected := false
+	stsFeatureGateEnabled := a.STSFeatureGateEnabled()
+	if stsFeatureGateEnabled {
+		stsDetected, err = utils.IsTimedTokenCluster(a.Client, logger)
+		if err != nil {
+			return err
+		}
 	}
-
-	switch credentialsRootSecret.Annotations[constants.AnnotationKey] {
-	case constants.InsufficientAnnotation:
-		msg := "cloud credentials insufficient to satisfy credentials request"
-		logger.Error(msg)
-		return &actuatoriface.ActuatorError{
-			ErrReason: minterv1.InsufficientCloudCredentials,
-			Message:   msg,
+	if stsFeatureGateEnabled && stsDetected {
+		if a.Codec == nil {
+			return fmt.Errorf("invalid codec, nil value")
```

Comment:
This seems like a programmer error , can it ever happen normally in runtime?

Reply:
yeah it seems like this would just catch a test setup error, because it seems like not getting a NewCodec() is handled elsewhere at runtime. Will remove this in upcoming commit.